src/scripts/reducers/UploadReducer.js

```diff
@@ -74,4 +86,4 @@ const reducer = {
   }
 }
 
-export default handleActions(reducer, fromJS(new UploadRecord()))
+export default handleActions(reducer, Map({}))
```

Comment:
Why is this no longer an `UploadRecord`?

Reply:
because it is a mapping of ids to Uploadrecords.
So this line looks good to me.
What looks fishy to me is the fact that all functions in the reducers are declared to operate on UploadREcords ( `state: UploadRecord`) but in fact do not

Reply:
Yes, this reducer looks pretty much screwed up.

I think we should keep `UploadRecord` and it should itself have a prop of `uploadedVideos` (or even just `videos`) that is this map of ids to `VideoRecord`